introducing-pgroll.mdx

```diff
@@ -0,0 +1,119 @@
+
+![pgroll virtual schemas during a migration](images/pgroll-migration-schemas.png)
+
+This defines the migration for setting a column as `NOT NULL` using `pgroll`. Additional operations can be included in the same migration, but we'll focus on this one for simplicity:
+
+```json
+{
+  "name": "review_not_null",
+  "operations": [
+    {
+      "alter_column": {
+        "table": "reviews",
+        "column": "review",
+        "not_null": true,
+        "up": "(SELECT CASE WHEN review IS NULL THEN product || ' is good' ELSE review END)",
+        "down": "review"
+      }
+    }
+  ]
+}
+```
+
+This is executed by running this simple command:
+
+```bash
+pgroll start review_not_null.json
+```
+
+`pgroll` will perform all the necessary steps to make this change available without disrupting existing client applications (the expand step from the [expand/contract pattern](https://openpracticelibrary.com/practice/expand-and-contract-pattern/)). In this particular example, it will:
+
+- Create a new column as a copy of the `review` column that respects the `NOT NULL` constraint.
+- Add the `NOT NULL` constraint to the new column, using the `NOT VALID` clause to ensure that data access doesn’t block due to this statement, avoiding unexpected downtime.
+- Backfill all existing values from the old column into the new one, upgrading the data in the cases where the value was `NULL`, based on the user-defined function. Of course, backfill happens in batches, avoiding excessively large updates that could also block the database.
+- Set up a trigger function so new inserts & updates get automatically copied between the new and the old columns.
+- Keep the old column working at all times, ensuring the migration can be rolled back instantly if needed.
+- Create a new view of the schema that hides the old column and promotes the new one in its place, exposing this way the new version of the schema while the old one keeps working.
+
+Typically, all of these steps would require manual execution to achieve a complex update like this. However, `pgroll` streamlines the process by performing all of them with a single command ✨.
+
+Once you are happy with the new version of the schema and all clients have been updated to use it, the old schema will no longer be accessed. It’s time to complete the migration (contract the schema) to get rid of the old column & triggers:
+
+```json
+pgroll complete
+```
+
+For more complex examples, there is a wide range of schema migration samples in our [examples](https://github.com/xataio/pgroll/tree/main/examples) and [docs](https://github.com/xataio/pgroll/tree/doc/document-alter-column-operation/docs).
+
+## What’s next?
+
+Today, we are rolling out the first version of `pgroll` and we are looking forward to your feedback!  We plan to continue developing `pgroll` and exploring how it can make it easier and safer for Xata users to interact with their schema.
```

Comment:
Small suggestion

```suggestion
Today, we are rolling out the first version of `pgroll` and we are looking forward to your feedback!  We plan to continue developing `pgroll` and exploring how it can make it easier and safer for Xata users to evolve their schema.
```